rllib/algorithms/algorithm.py

```diff
@@ -1547,7 +1547,8 @@ def set_weights(self, weights: Dict[PolicyID, dict]):
     def add_policy(
         self,
         policy_id: PolicyID,
-        policy_cls: Type[Policy],
+        policy_cls: Optional[Type[Policy]] = None,
```

Comment:
Simplified the `Algorithm.add_policy()` method by only using the new WorkerSet APIs. No more micro-handling individual workers policy_maps here.